src/main/java/com/google/code/or/io/util/ActiveBufferedInputStream.java

```diff
@@ -174,7 +174,7 @@ public int read(byte b[], int off, int len) throws IOException {
         	while (this.ringBuffer.isEmpty()) {
         		if(this.exception != null) throw this.exception;
             	this.bufferNotEmpty.awaitUninterruptibly();
-            	if(this.closed.get()) throw new EOFException();
+            	if(this.closed.get() && this.ringBuffer.isEmpty()) throw new EOFException();

```

Comment:
@dadah89 I fixed a race condition here that was triggering bad `EOFException`s to be thrown